core/src/main/java/io/questdb/log/LogAlertSocketWriter.java

```diff
@@ -0,0 +1,303 @@
+        this.clock = clock;
+        this.alertsSourceQueue = alertsSrc;
+        this.writeSequence = writeSequence;
+        this.level = level & ~(1 << Numbers.msb(LogLevel.ADVISORY)); // switch off ADVISORY
+    }
+
+    @Override
+    public void bindProperties() {
+        int nBufferSize = LogAlertSocket.OUT_BUFFER_SIZE;
+        if (bufferSize != null) {
+            try {
+                nBufferSize = Numbers.parseIntSize(bufferSize);
+            } catch (NumericException e) {
+                throw new LogError("Invalid value for bufferSize");
+            }
+        }
+        socket = new LogAlertSocket(alertTargets, nBufferSize);
+        loadLogAlertTemplate();
+        socket.connect();
+    }
+
+    @Override
+    public void close() {
+        if (socket != null) {
+            socket.close();
+        }
+    }
+
+    @Override
+    public boolean runSerially() {
+        return writeSequence.consumeAll(alertsSourceQueue, alertsProcessor);
+    }
+
+    @VisibleForTesting
+    void setBufferSize(String bufferSize) {
+        this.bufferSize = bufferSize;
+    }
+
+    @VisibleForTesting
+    void setLocation(String location) {
+        this.location = location;
+    }
+
+    @VisibleForTesting
+    String getLocation() {
+        return location;
+    }
+
+    @VisibleForTesting
+    void setAlertTargets(String alertTargets) {
+        this.alertTargets = alertTargets;
+    }
+
+    @VisibleForTesting
+    String getAlertTargets() {
+        return socket.getAlertTargets();
+    }
+
+    @VisibleForTesting
+    HttpLogAlertBuilder getAlertBuilder() {
+        return alertBuilder;
+    }
+
+    @VisibleForTesting
+    int getBufferSize() {
+        return socket.getOutBufferSize();
+    }
+
+    private void loadLogAlertTemplate() {
+        final long now = clock.getTicks();
+        if (location.isEmpty()) {
+            location = DEFAULT_ALERT_TPT_FILE;
+        }
+        location = dollar$.resolveEnv(location, now).toString(); // location may contain dollar expressions
+
+        // read template, resolve env vars within (except $ALERT_MESSAGE)
+        boolean wasRead = false;
+        try (InputStream is = LogAlertSocketWriter.class.getResourceAsStream(location)) {
+            if (is != null) {
+                dollar$.resolve(CharSequenceView.of(is), now, alertProps);
+                wasRead = true;
+            }
+        } catch (IOException e) {
+            // it was not a resource ("/resource_name")
+        }
+        if (!wasRead) {
+            dollar$.resolve(
+                    readFile(
+                            location,
+                            socket.getInBufferPtr(),
+                            socket.getInBufferSize(),
+                            ff
+                    ),
+                    now,
+                    alertProps
+            );
+        }
+        // consolidate/check/load template to the outbound socket buffer
+        dollar$.resolve(dollar$.toString(), now, alertProps);
```

Comment:
if there are dates in the template and server is running for a week, the date in the JSON will be that of the startup and not the alert

Reply:
that is a limitation, the original requirements suggest a template that does not contain dates (see above), beyond the timestamp prefix that comes in the actual log we are trying to alert for. 
If TemplateParser allows for expressions such as `${date: <format>}` it is because that code was first used for resolving rolling log file names. 

Would a scenario like this make sense:

```
[
  {
    "Status": "firing",
    "Labels": {
      "alertname": "QuestDbInstanceLogs",
      "category": "application-logs",
      "severity": "critical",
      "orgid": "GLOBAL",
      "service": "QuestDB",
      "namespace": "GLOBAL",
      "cluster": "GLOBAL",
      "instance": "GLOBAL",
      "date": "${date: y}"
    },
    "Annotations": {
      "description": "ERROR/GLOBAL/GLOBAL/GLOBAL/GLOBAL",
      "message": "${ALERT_MESSAGE}"
    }
  }
]
```
if so, I will need a redesign.

Reply:
Yes, I would imagine scenario like this could make sense. @rstreics what do you think? Do we need date as label ?

Its not much of a re-design, just don't consolidate the components. Use them as a list, start from topmost _static_ .

Reply:
ok, I can do that, but it will impact the performance of the onLogRecord

now:
- the outbound socket buffer contains a preloaded header (http header + static message template until the $ of ${ALERT_MESSAGE} in the template
- so we simply rewind to that point, append the message, then append the static tailing part

after this:
- In addition to ${ALERT_MESSAGE}, no ${date: <format>} expression will be resolved, until the invocation of onLogRecord
- at this point we resolve everything pumping it down the socket, until we find the ALERT_MESSAGE in the template, point at which we send the log record, followed by whatever is left unresolved beyond the ALERT_MESSAGE

I will double check whether there is an actual use case

Reply:
I am giving it a go now, I think I can make it efficient